crux-bench/time-series-bench/src/time_series_bench/ts_devices.clj

```diff
@@ -0,0 +1,300 @@
+                        :last-tx (crux/submit-tx node (vec (for [{:keys [reading/time] :as reading-doc} chunk]
+                                                             [:crux.tx/put reading-doc time])))})
+                     {:op-count (count info-tx-ops)}))))))
+;; 10 most recent battery temperature readings for charging devices
+;; SELECT time, device_id, battery_temperature
+;; FROM readings
+;; WHERE battery_status = 'charging'
+;; ORDER BY time DESC LIMIT 10;
+
+;; time                   | device_id  | battery_temperature
+;; -----------------------+------------+---------------------
+;; 2016-11-15 23:39:30-05 | demo004887 |                99.3
+;; 2016-11-15 23:39:30-05 | demo004882 |               100.8
+;; 2016-11-15 23:39:30-05 | demo004862 |                95.7
+;; 2016-11-15 23:39:30-05 | demo004844 |                95.5
+;; 2016-11-15 23:39:30-05 | demo004841 |                95.4
+;; 2016-11-15 23:39:30-05 | demo004804 |               101.6
+;; 2016-11-15 23:39:30-05 | demo004784 |               100.6
+;; 2016-11-15 23:39:30-05 | demo004760 |                99.1
+;; 2016-11-15 23:39:30-05 | demo004731 |                97.9
+;; 2016-11-15 23:39:30-05 | demo004729 |                99.6
+;; (10 rows)
+
+
+(defn test-10-most-recent-battery-readings-for-charging-devices [node]
+  (try
+    (let [start-time (System/currentTimeMillis)
+          query '{:find [time device-id battery-temperature]
+                  :where [[r :reading/time time]
+                          [r :reading/device-id device-id]
+                          [r :reading/battery-temperature battery-temperature]]
+                  :order-by [[time :desc] [device-id :desc]]
+                  :limit 10}
+          query-result (crux/q (crux/db node) query)
+          end-time (System/currentTimeMillis)
+          successful? (= query-result
+                         [[#inst "2016-11-15T20:19:30.000-00:00" :device-info/demo000999 88.7]
+                          [#inst "2016-11-15T20:19:30.000-00:00" :device-info/demo000998 93.1]
+                          [#inst "2016-11-15T20:19:30.000-00:00" :device-info/demo000997 90.7]
+                          [#inst "2016-11-15T20:19:30.000-00:00" :device-info/demo000996 92.8]
+                          [#inst "2016-11-15T20:19:30.000-00:00" :device-info/demo000995 91.9]
+                          [#inst "2016-11-15T20:19:30.000-00:00" :device-info/demo000994 92.0]
+                          [#inst "2016-11-15T20:19:30.000-00:00" :device-info/demo000993 92.8]
+                          [#inst "2016-11-15T20:19:30.000-00:00" :device-info/demo000992 87.6]
+                          [#inst "2016-11-15T20:19:30.000-00:00" :device-info/demo000991 93.1]
+                          [#inst "2016-11-15T20:19:30.000-00:00" :device-info/demo000990 89.9]])]
+      (println (json/write-str {:crux.bench/bench-type ::recent-battery-readings
+                                ::query query
```

Comment:
these keys aren't specific to this namespace, more to `crux-bench` as a whole - if we can use a consistent name, we can compare them between benchmarks, as and when we have more

Reply:
`clojure.data.json` strips namespaces anyway, so the `::` here is totally irrelevant, but point taken